pkg/expapi/types.go

```diff
@@ -27,3 +27,37 @@ support is experimental.
 */
 
 package expapi
+
+import "k8s.io/kubernetes/pkg/api"
+
+// ScalerSpec describes the attributes a Scaler subresource
+type ScalerSpec struct {
+	// Replicas is the number of desired replicas. This is a pointer to distinguish between explicit zero and unspecified.
+	Replicas *int `json:"replicas,omitempty" description:"number of replicas desired; defaults to 1; see http://releases.k8s.io/HEAD/docs/user-guide/replication-controller.md#what-is-a-replication-controller"`
+}
+
+// ScalerStatus represents the current status of a Scaler subresource.
+type ScalerStatus struct {
+	// Replicas is the number of actual replicas.
+	Replicas int `json:"replicas" description:"most recently oberved number of replicas; see http://releases.k8s.io/HEAD/docs/user-guide/replication-controller.md#what-is-a-replication-controller"`
+
+	// Selector is a label query over pods that should match the replicas count.
+	Selector map[string]string `json:"selector,omitempty" description:"label keys and values that must match in order to be controlled by this replication controller, if empty defaulted to labels on Pod template; see http://releases.k8s.io/HEAD/docs/user-guide/labels.md#label-selectors"`
+}
+
+// Scaler subresource, applicable to ReplicationControllers and (in future) Deployment.
+type Scaler struct {
```

Comment:
Nit: I prefer `Scale` over `Scaler`. Our API is declarative, and creating API types that sound like system components has caused a fair amount of confusion and awkwardness in the past.